app/controllers/agent/ConfirmClientVrnController.scala

```diff
@@ -68,16 +68,33 @@ class ConfirmClientVrnController @Inject()(val messagesApi: MessagesApi,
   }
 
   def redirect: Action[AnyContent] = authenticate {
+
     implicit user =>
-      user.session.get(SessionKeys.redirectUrl) match {
-        case Some(redirectUrl) =>
+
+      val redirectUrl = user.session.get(SessionKeys.redirectUrl)
+      val hasVerifiedAgentEmail: Boolean = user.session.get(SessionKeys.verifiedAgentEmail).isDefined
+      val manageVatUrl = appConfig.manageVatCustomerDetailsUrl
+
+       redirectUrl match {
+
+        case Some(changeUrl) if changeUrl.contains("/vat-through-software/account")  =>
           user.session.get(SessionKeys.preference) match {
-            case Some(_) => Redirect(redirectUrl).removingFromSession(SessionKeys.redirectUrl)
+            case Some("yes") if hasVerifiedAgentEmail =>
+              Redirect(manageVatUrl).removingFromSession(SessionKeys.redirectUrl)
+            case Some("yes") if !hasVerifiedAgentEmail =>
+              Redirect(controllers.agent.routes.ConfirmEmailController.isEmailVerified())
+            case Some("no") =>
```

Comment:
Could you add some tests for each of these `Some` cases? Scoverage is saying they aren't tested at the moment.